app/src/main/java/com/example/android/chemapp/MainActivity.java

```diff
@@ -145,11 +156,45 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
     public void onSearchResultClicked(ChemElement repo) {
         Intent intent = new Intent(this, ElementDetailActivity.class);
         intent.putExtra(ElementDetailActivity.EXTRA_CHEM_ELEMENT, repo);
+        //Log.d(TAG, "onSearchResultClicked: " + repo.name);
         startActivity(intent);
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.toolbar_menu, menu);
+
+        MenuItem menuItem = menu.findItem(R.id.action_search);
+        SearchView searchView = (SearchView) menuItem.getActionView();
+        searchView.setOnQueryTextListener(this);
+        return true;
+    }
+
     private void getPeriodicTable(String searchQuery) {
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
         mViewModel.loadSearchResults(searchQuery);
     }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        String userInput = newText.toLowerCase();
+        List<ChemElement> newList = new ArrayList<>();
+
+        for (int i = 0; i < 118 ; i++) {
```

Comment:
Could we change the hard-coded 118 to results.size() in case it doesn't have 118 entries for some reason

Reply:
good point! when I wrote the loop I just looked up how many elements there are on the periodic table. I'll be sure to change that.